lib/Db/OldPasswordMapper.php

```diff
@@ -63,4 +63,30 @@ public function getLatestPassword($uid) {
 		}
 		return $passwords[0];
 	}
+
+	public function getPasswordsAboutToExpire($maxTimestamp) {
+		$oldPasswords = [];
+
+		$query = "select `f`.`*` from (";
```

Comment:
I'd prefer using a simpler query like `SELECT uid, MAX(change_time) as `maxtime` FROM`*PREFIX*user_password_history` GROUP BY uid HAVING MAX(change_time) < ? `

It was pointed to me that the reason for this is because the `OldPassword` object requires the original fields to be present in the select, so this JOIN is here just to achieve this.

I suggested skipping the OldPassword object here and just returning a list of users. This wouldn't match well with the "ObjectMapper" concept. So maybe we should move this method elsewhere ?

Our problem currently is to extend this query to also cover cases where no change_time is set for https://github.com/owncloud/password_policy/issues/24

Reply:
@jvillafanez how about adding a new method on `OldPassword` that lets you create an instance with a given id ? Then here you could simplify the query as discussed instead of having to contraint it to `OldPassword::fromRow()`

Reply:
it should be possible to include all information in the original query somehow

let's continue for now and if needed we can ask @butonic to help us optimize this

unit tests will prove that it still works regardless